src/storage/src/hummock/iterator/merge_inner.rs

```diff
@@ -88,22 +90,122 @@ impl<'a, const DIRECTION: usize> MergeIteratorInner<'a, DIRECTION> {
 
         self.heap = self
             .unused_iters
-            .drain_filter(|i| i.is_valid())
-            .map(Node)
+            .drain_filter(|i| i.iter.is_valid())
             .collect();
     }
 }
 
 #[async_trait]
-impl<const DIRECTION: usize> HummockIterator for MergeIteratorInner<'_, DIRECTION> {
-    async fn next(&mut self) -> HummockResult<()> {
+/// The behaviour of `next` of order aware merge iterator is different from the normal one, so we
+/// extract this trait.
+trait MergeIteratorNext<'a> {
```

Comment:
Is this `'a` necessary?

Reply:
Seems unnecessary if we are doing static dispatch. Have removed it.